alchemy/bin/commands/deploy.ts

```diff
@@ -0,0 +1,216 @@
+import { exec } from "node:child_process";
+import { existsSync, watch, readdirSync, statSync } from "node:fs";
+import { resolve, dirname, join } from "node:path";
+import { promisify } from "node:util";
+import { log, spinner } from "@clack/prompts";
+import pc from "picocolors";
+import { detectPackageManager } from "../services/package-manager.ts";
+
+const execAsync = promisify(exec);
+
+export interface DeployInput {
+  path?: string;
+  quiet?: boolean;
+  read?: boolean;
+  destroy?: boolean;
+  stage?: string;
+  watch?: boolean;
+}
+
+export async function deployAlchemy(input: DeployInput = {}) {
+  const {
+    path = process.cwd(),
+    quiet,
+    read,
+    destroy,
+    stage,
+    watch: watchMode,
+  } = input;
+
+  // Check for alchemy.run.ts or alchemy.run.js
+  const runTsPath = resolve(path, "alchemy.run.ts");
+  const runJsPath = resolve(path, "alchemy.run.js");
+
+  let runFile: string | null = null;
+  if (existsSync(runTsPath)) {
+    runFile = runTsPath;
+  } else if (existsSync(runJsPath)) {
+    runFile = runJsPath;
+  }
+
+  if (!runFile) {
+    log.error(
+      pc.red(
+        "No alchemy.run.ts or alchemy.run.js file found in the current directory.",
+      ),
+    );
+    log.info("Create an alchemy.run.ts file to define your infrastructure.");
+    process.exit(1);
+  }
+
+  // Detect package manager
+  const packageManager = detectPackageManager(path);
+
+  // Build command arguments
+  const args: string[] = [];
+  if (quiet) args.push("--quiet");
+  if (read) args.push("--read");
+  if (destroy) args.push("--destroy");
+  if (stage) args.push(`--stage ${stage}`);
+
+  const argsString = args.join(" ");
+
+  // Determine the command to run based on package manager and file extension
+  let command: string;
+  const isTypeScript = runFile.endsWith(".ts");
+
+  switch (packageManager) {
+    case "bun":
+      command = `bun ${runFile} ${argsString}`;
+      break;
+    case "deno":
+      command = `deno run -A ${runFile} ${argsString}`;
+      break;
+    case "pnpm":
+      command = isTypeScript
+        ? `pnpm tsx ${runFile} ${argsString}`
+        : `pnpm node ${runFile} ${argsString}`;
+      break;
+    case "yarn":
+      command = isTypeScript
+        ? `yarn tsx ${runFile} ${argsString}`
+        : `yarn node ${runFile} ${argsString}`;
+      break;
+    default:
+      command = isTypeScript
+        ? `npx tsx ${runFile} ${argsString}`
+        : `node ${runFile} ${argsString}`;
+      break;
+  }
+
+  // Execute the deploy
+  async function executeDeploy() {
+    const s = spinner();
+    s.start(`Executing: ${command}`);
+
+    try {
+      const { stdout, stderr } = await execAsync(command, {
```

Comment:
We could also dynamically import the file, not sure if that's better. It would allow us to introspect the state for later use cases.

Reply:
Hmm but might not work as well for different runtimes. Ok new process makes more sense.